src/prettytable/prettytable.py

```diff
@@ -1454,6 +1454,7 @@ def _get_options(self, kwargs: Mapping[str, Any]) -> OptionsType:
 
     def set_style(self, style: TableStyle) -> None:
         self._style = style
+        self._set_default_style()
         if style == TableStyle.DEFAULT:
             self._set_default_style()
         elif style == TableStyle.MSWORD_FRIENDLY:
```

Comment:
We can remove this, because we've just called `self._set_default_style`:

```suggestion
        if style == TableStyle.MSWORD_FRIENDLY:
```